oe_theme.theme

```diff
@@ -1522,3 +1523,88 @@ function oe_theme_preprocess_page(array &$variables): void {
     'id' => ['main-content'],
   ]);
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function oe_theme_preprocess_oe_organisation(&$variables): void {
+  $cacheability = CacheableMetadata::createFromRenderArray($variables);
+
+  $entity = &$variables['entity'];
+  // Prepare fields.
+  $fields_values = ['name' => 'name', 'oe_acronym' => 'acronym'];
+  foreach ($fields_values as $field => $pattern_key) {
+    if ($entity->get($field)->isEmpty()) {
+      $variables[$pattern_key] = '';
+      continue;
+    }
+    $variables[$pattern_key] = $entity->get($field)->value;
+  }
+
+  // Extract the image if present.
+  $variables['logo'] = '';
+  if (!$entity->get('oe_logo')->isEmpty()) {
+    /** @var \Drupal\file\Plugin\Field\FieldType\FileItem $image_item */
+    $media_image = $entity->get('oe_logo')->entity;
+
+    if (!$media_image->get('oe_media_image')->isEmpty()) {
+      $image_item = $media_image->get('oe_media_image')->first();
+      $variables['logo'] = ImageValueObject::fromImageItem($image_item);
+      // Add cache data from media image and file.
+      $file = $image_item->get('entity')->getValue();
+      $cacheability->addCacheableDependency($media_image);
+      $cacheability->addCacheableDependency($file);
+    }
+  }
+
+  // Set contact page url and icon.
+  $variables['contact_page_url'] = '';
+  if (!$entity->get('oe_contact_url')->isEmpty()) {
+    $link = $entity->get('oe_contact_url')->first()->getValue();
+    if (UrlHelper::isExternal($link['uri'])) {
+      $variables['contact_page_url'] = $link['uri'];
+    }
+    else {
+      $variables['contact_page_url'] = Url::fromUri($link['uri'])->toString();
+    }
+  }
+
+  // Process contact item.
+  $variables['organisation_details'] = [];
+  $contact_fields = ['oe_address', 'oe_website'];
+  foreach ($contact_fields as $field) {
+    if (empty($variables['content'][$field]['#title'])) {
+      continue;
+    }
+    $variables['organisation_details'][] = [
+      'label' => $variables['content'][$field]['#title'],
+      'body' => [
+        '#label_display' => 'hidden',
+      ] + $variables['content'][$field],
+    ];
+  }
+
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function oe_theme_preprocess_pattern_organisation_teaser(array &$variables): void {
+  $contact_page_url = $variables['contact_page_url'];
+
+  // Define icon based on the url type.
+  $variables['contact_page']['url'] = $contact_page_url;
+  if (UrlHelper::isExternal($contact_page_url)) {
+    $variables['contact_page']['icon'] = 'external';
+    $variables['contact_page']['url'] = $contact_page_url;
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function oe_theme_preprocess_pattern_organisation_teaser__preview(array &$variables): void {
+  if (!empty($variables['logo'])) {
```

Comment:
We know logo won't be empty as we are rendering a pattern preview, so we can remove this if.